### PR TITLE
migrate from rusoto_s3 to official aws-s3-sdk

### PR DESCRIPTION
**Pull request by syphar**
- rusoto is in maintenance mode : https://github.com/rusoto/rusoto/releases/tag/rusoto-v0.48.0 
- we probably only need either this PR or #1733

### CODE REVIEW

**Comment by syphar** [comment]
r? @jyn514 @Nemo157  

the official SDK has its own retry logic so I removed ours.

File: src/storage/s3.rs [hunk 1]
```diff
@@ -56,19 +29,20 @@ impl S3Backend {
...
-                .as_deref()
-                .map(|endpoint| Region::Custom {
-                    name: config.s3_region.name().to_string(),
-                    endpoint: endpoint.to_string(),
-                })
-                .unwrap_or_else(|| config.s3_region.clone()),
-        );
+        let shared_config = runtime.block_on(aws_config::load_from_env());
+        let mut config_builder = aws_sdk_s3::config::Builder::from(&shared_config)
+            .retry_config(RetryConfig::new().with_max_attempts(5))
```
Comment (jyn514) [inline src/storage/s3.rs:34]: We had 3 max attempts before - can we keep using that?
```suggestion
            .retry_config(RetryConfig::new().with_max_attempts(3))
```

File: src/storage/s3.rs [hunk 2]
```diff
@@ -261,37 +222,34 @@ impl<'a> StorageTransaction for S3StorageTransaction<'a> {
...
-                        version_id: None,
-                    })
-                    .collect::<Vec<_>>();
-
-                if !to_delete.is_empty() {
+                if list.key_count() > 0 {
+                    let to_delete = Delete::builder()
+                        .set_objects(Some(
+                            list.contents
+                                .expect("didn't get context even though but key_count was > 0")
```
Comment (jyn514) [inline src/storage/s3.rs:236]:
```suggestion
                                .expect("didn't get contents even though key_count was > 0")
```

**Review by jyn514 — COMMENTED** [review]
This is awesome, thanks! I've been meaning to look for a maintained s3 library for a while :)

**Review by jyn514 — APPROVED** [review]

**Pull request merged**